bcachefs/bcachefs.py

```diff
@@ -1011,11 +1031,14 @@ def __init__(self, fs: _Bcachefs):
         self._deleted = set()
 
     def __next__(self):
-        dirent = DirEnt(*super(BcachefsIterDirEnt, self).__next__())
+        dirent = super(BcachefsIterDirEnt, self).__next__()
         while (
             not dirent.inode
             or (dirent.parent_inode, dirent.name) in self._deleted
         ):
             self._deleted.add((dirent.parent_inode, dirent.name))
-            dirent = DirEnt(*super(BcachefsIterDirEnt, self).__next__())
+            dirent = super(BcachefsIterDirEnt, self).__next__()
```

Comment:
In Python 3, `super().whatever` is equivalent to `super(ClassName, self)` except without the redundancy. The duplication of the class name threatens to become a foot-gun whenever code is copy-pasted.

Reply:
Ya, and maybe I should instead use directly `BcachefsIter.__next__(self)`. I'm usually not inclined to encourage the use of super() chaining which I believe is prone to bugs when used inconsistently (like in this case where the code is not thought to support chaining of `__next__()` calls). My mind is apparently still on vacations